### PR TITLE
Add support for lxd snap socket, drop juju wrapper in snap

### PR DESCRIPTION
**Pull request by nskaggs**
## Description of change

Adds support for snap socket of LXD, and prefers it. Drops the wrapper in snap that previously allowed this support.

## QA steps

sudo snap install lxd
juju bootstrap lxd
Juju should bootstrap using lxd socket

sudo apt install lxd (now 2 lxd's are installed)
juju bootstrap lxd
Juju should still bootstrap using lxd socket

sudo snap remove lxd (now only deb lxd installed)
juju bootstrap lxd
Juju should bootstrap using debian socket

sudo apt remove lxd (no lxd installed)
juju bootstrap lxd
Juju should prompt to install lxd snap

## Bug reference

Fixes LP [#1743719](https://bugs.launchpad.net/juju/+bug/1743719)

### CODE REVIEW

File: tools/lxdclient/client.go [hunk 1]
```diff
@@ -215,6 +216,19 @@ func isSupportedAPIVersion(version string) bool {
 func newRawClient(remote Remote) (*lxd.Client, error) {
 	host := remote.Host
 
+	// LXD socket is different depending on installation method
```
Comment (jameinel) [inline tools/lxdclient/client.go]: We need to have a check that we don't set LXD_DIR if it is already set.

**Comment by nskaggs** [comment]
[Juju should support lxd snap socket](https://trello.com/c/XKToyHn0/156-juju-should-support-lxd-snap-socket)

File: tools/lxdclient/client.go [hunk 2]
```diff
@@ -220,13 +220,21 @@ func newRawClient(remote Remote) (*lxd.Client, error) {
...
+	// The current lxdclient code ONLY looks at LXD_DIR, as it has no understanding
+	// of snaps. This forces us to set LXD_DIR to set the socket path.
+	// This should be updated once lxdclient dependency is updated
+	if envSocket != "" {
+		logger.Debugf("Using environment LXD_DIR as socket path: %q", envSocket)
 	} else {
-		logger.Debugf("LXD snap socket not found, falling back to debian socket")
-		os.Setenv("LXD_DIR", debianSocket)
+		if _, err := os.Stat(snapSocket); err == nil {
+			logger.Debugf("Using LXD snap socket")
```
Comment (jameinel) [inline tools/lxdclient/client.go]: is it worth printing the paths here
"Using LXD snap socket: %s", snapSocket)

anyway, either way, LGTM

Comment (nskaggs) [reply tools/lxdclient/client.go]: I like to be explicit when logging, so yes, let's print the paths for all of them in debug.

**Comment by nskaggs** [comment]
$$merge$$

**Pull request merged**